pyccel/ast/variable.py

```diff
@@ -917,27 +891,34 @@ def __repr__(self):
 
     def __getitem__(self, *args):
 
+        if self.class_type.container_rank < len(args):
+            raise IndexError('Rank mismatch.')
+
         if len(args) == 1 and isinstance(args[0], (tuple, list)):
             args = args[0]
 
-        if self.rank < len(args):
-            raise IndexError('Rank mismatch.')
-
-        new_indexes = []
-        j = 0
-        base = self.base
-        for i in self.indices:
-            if isinstance(i, Slice) and j<len(args):
-                if i.step == 1 or i.step is None:
-                    incr = args[j]
-                else:
-                    incr = PyccelMul(i.step, args[j], simplify = True)
-                if i.start != 0 and i.start is not None:
-                    incr = PyccelAdd(i.start, incr, simplify = True)
-                i = incr
-                j += 1
-            new_indexes.append(i)
-        return IndexedElement(base, *new_indexes)
+        if self._is_slice:
+            new_indexes = []
+            j = 0
+            base = self.base
+            for i in self.indices:
+                if isinstance(i, Slice) and j<len(args):
+                    current_arg = args[j]
+                    if isinstance(current_arg, Slice):
+                        raise NotImplementedError("Can't extract a slice from a slice")
```

Comment:
`a[1:][2:]`

Reply:
Interesting, I had not realized that we were not supporting this. Is there a fundamental reason?

Reply:
Mostly because it is a pain :sweat_smile: . It would be possible if we create a temporary but the way we handle:
`a[1::2][2]` is to translate it to `a[1+2*2]`. A slice can't be easily handled like this

Reply:
All right. Do we have a word about this in the user's documentation?

Reply:
No. Which bit specifically do you think we should mention? The fact that indices and slices are combined when used in the same expression? Or the fact that repeated slicing is not possible? Both are corner cases that I don't think we are likely to find in the wild. The following equivalent code works:
```python
b = a[1:]
b[2:]
```

Reply:
In the user docs I would mention the latter, i.e. that repeated slicing is not possible.

Reply:
We could also add that we think one should not use repeated slicing anyway, as it makes the code unnecessary complicated to read.

Reply:
Done 06bf4b6e3